apikana-maven-plugin/src/main/java/org/swisspush/apikana/AbstractApikanaMojo.java

```diff
@@ -172,23 +222,121 @@ protected boolean isPom() {
+            devDependencies.put("apikana-defaults", apikanaDefaultsVersion);
+            devDependencies.put("typescript", typescriptVersion);
+            final Map<String, Object> customConfig = (Map) pack.merge("customConfig", new LinkedHashMap<>(), (oldVal, newVal) -> oldVal);
+            customConfig.put("type", type);
+            customConfig.put("domain", domain);
+            customConfig.put("author", getAuthor(apiSpec));
+            customConfig.put("namespace", namespace);
+            customConfig.put("shortname", shortname);
+            customConfig.put("projectName", mavenProject.getArtifactId());
+            customConfig.put("title", getTitle(apiSpec));
             final List<String> plugins = new ArrayList<>();
             plugins.add("maven");
             plugins.add("readme");
             customConfig.put("plugins", plugins);
+            customConfig.put("javaPackage", javaPackage);
+            customConfig.put("mavenGroupId", mavenProject.getGroupId());
+            final Map<String, Object> avro = (Map) customConfig.merge("avro", new LinkedHashMap<>(), (oldVal, newVal) -> oldVal);
+            avro.put("enumAsString", true);
+            customConfig.put("avro", avro);
             pack.put("customConfig", customConfig);
-            devDependencies.put("apikana", version);
-            devDependencies.put("apikana-defaults", apikanaDefaultsVersion);
         });
     }
 
+    private String getTitle(Map<String, Object> apiSpec) {
+        Map<String, Object> info = (Map<String, Object>) apiSpec.get("info");
+        return (String) info.get("title");
+    }
+
+    private String getAuthor(Map<String, Object> apiSpec) {
+        Map<String, Object> info = (Map<String, Object>) apiSpec.get("info");
+        Map<String, Object> contact = (Map<String, Object>) info.get("contact");
+
+        StringBuilder authorBuilder = new StringBuilder();
+
+        if (contact.get("name") != null) {
+            authorBuilder.append(contact.get("name"));
+        }
+
+        if (contact.get("email") != null) {
+            if (authorBuilder.length() > 0) {
+                authorBuilder.append(" <")
+                        .append(contact.get("email"))
+                        .append(">");
+            } else {
+                authorBuilder.append(contact.get("email"));
+            }
+        }
+
+        if (contact.get("url") != null) {
+            if (authorBuilder.length() > 0) {
+                authorBuilder.append(" (")
+                        .append(contact.get("url"))
+                        .append(")");
+            } else {
+                authorBuilder.append(contact.get("url"));
+            }
+        }
+
+        return authorBuilder.toString();
+    }
+
+    private Map<String, Object> parseApiSpec() {
+        try {
+            getLog().info(String.format("basedir: %s", mavenProject.getBasedir().getAbsolutePath()));
+            getLog().info(String.format("api: %s", api));
```

Comment:
(*trivial*) To me it feels using *String.format* does not make any sense here.

I assume this logger just takes a string. If my assumption is correct, this is not a problem and we could also keep it as-is. BUT if this is a slf4 logger, then I strongly suggest to use parameterized logging instead.

Reply:
I reworked that logging block so that I can remove `String.format()` without losing the shortness of the string value.

Reply:
Somehow I cannot see any change here too. What I meant is something like:

```patch
-  getLog().info(String.format("basedir: %s", mavenProject.getBasedir().getAbsolutePath()));
+  getLog().info("basedir: "+ mavenProject.getBasedir().getAbsolutePath());
```
And
```patch
-  getLog().info(String.format("api: %s", api));
+  getLog().info("api: "+ api);
```

As using "*String.format*" has zero benefit here.